pkg/internal/controller/controller.go

```diff
@@ -83,8 +85,12 @@ type Controller struct {
 	// startWatches maintains a list of sources, handlers, and predicates to start when the controller is started.
 	startWatches []watchDescription
 
-	// Log is used to log messages to users during reconciliation, or for example when a watch is started.
-	Log logr.Logger
+	// LogConstructor is used to construct a logger to then log messages to users during reconciliation,
+	// or for example when a watch is started.
+	LogConstructor func(request *reconcile.Request) logr.Logger
+
+	// RandSource is used to generate reconcileIDs for logging.
+	RandSource *rand.Rand
```

Comment:
Why is this configurable? Is there a valid usecase for wanting a custom rand source for the reconcile ID? And I really think the whole id generation would be easier if we used something like the uuid package

Reply:
I'll check the uuid package. It absolutely doesn't have to be configurable. I added it here because I didn't want to create a new RandSource for every reconcile. But when we move to uuid that's not necessary anymore.

Reply:
Has been dropped